nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -228,12 +249,19 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                     outgoingFlowFile = session.putAllAttributes(outgoingFlowFile, attributes);
                     session.getProvenanceReporter().receive(outgoingFlowFile, getURI(context));
                     session.transfer(outgoingFlowFile, REL_SUCCESS);
+                    sent++;
                 }
             }
 
             if (input != null) {
                 session.transfer(input, REL_ORIGINAL);
             }
+
+            if (sent == 0 && sendEmpty) {
+                FlowFile empty = session.create(input);
```

Comment:
Still tracking down whether there's a bug or intended behavior, but if `input == null` here it causes an NPE (looks like its due to changes in #2974 ). Just to be safe it should probably check for null here and just call `create()` instead. Would make a good unit test too (to show it works without an incoming flow file).

Reply:
Ok. Fixed that.